aws/iam.go

```diff
@@ -0,0 +1,66 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+)
+
+// List all IAM users in the AWS account and returns a slice of the UserNames
+// TODO: Implement exclusion by time filter
+func getAllIamUsers(session *session.Session, configObj config.Config) ([]*string, error) {
+	svc := iam.New(session)
+	input := &iam.ListUsersInput{}
+
+	var userNames []*string
+
+	// TODO: Probably use ListUsers together with ListUsersPages in case there are lots of users
+	output, err := svc.ListUsers(input)
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+
+	for _, user := range output.Users {
+		if config.ShouldInclude(*user.UserName, configObj.IAMUsers.IncludeRule.NamesRE, configObj.IAMUsers.ExcludeRule.NamesRE) {
+			userNames = append(userNames, user.UserName)
+		}
+	}
+
+	return userNames, nil
+}
+
+// TODO: This is only deleting the user but no associated resources
+// According to https://docs.aws.amazon.com/sdk-for-go/api/service/iam/#IAM.DeleteUser
+// "you must delete the items attached to the user manually, or the deletion fails"
```

Comment:
Unfortunately, this needs to be handled before we can merge this in as it is going to be super common for users to have stuff associated with it.

We should replicate https://github.com/gruntwork-io/prototypes/blob/master/aws-gc/aws-gc.py#L74 here.